vochain/state.go

```diff
@@ -561,6 +591,36 @@ func (v *State) incrementTreasurerNonce() error {
 	return v.Tx.DeepSet([]byte(TreasurerKey), tBytes, ExtraCfg)
 }
 
+func (v *State) SetTxCostByTxType(txType models.TxType, cost uint64) error {
+	key, ok := TxTypeToKey[txType]
+	if !ok {
+		return fmt.Errorf("txType %v shouldn't cost anything", txType)
+	}
+	return v.SetTxCost(key, cost)
+}
+
+func (v *State) SetTxCost(key string, cost uint64) error {
+	v.Tx.Lock()
+	defer v.Tx.Unlock()
+
+	costBytes := [8]byte{}
+	binary.LittleEndian.PutUint64(costBytes[:], cost)
+	return v.Tx.DeepSet([]byte(key), costBytes[:], ExtraCfg)
+}
+
+func (v *State) TxCostByTxType(txType models.TxType) (uint64, error) {
```

Comment:
Can we have the `isQuery` approach here ? Like we have in `Oracles()` or `Treasurer()`